### PR TITLE
Production logic

### PR DESCRIPTION
**Pull request by emirodriguezalc**

### CODE REVIEW

File: src/Building.h
Comment (d3uk4li0n) [inline src/Building.h:1]: Consider making "level" and "position" private members of the "Building" class and provide public getter methods to access them for encapsulation 

Consider making values like "Farmland" and "Mine" constants, as they probably will not change

**Review by d3uk4li0n — COMMENTED** [review]
Building.h: Consider making "level" and "position" private members of the "Building" class and provide public getter methods to access them for encapsulation

Consider making values like "Farmland" and "Mine" constants, as they probably will not change

**Review by staskashchenko — APPROVED** [review]
1)Building.h: Placing static constants inside a class can lead to linking problems if these constants are used in other files. Consider defining these constants outside of the class or using inline to avoid such problems.

2)You have the classes Building, Farmland, Mine, Quarry, and LumberMill defined in one header file. This can make the code difficult to maintain as it grows. Consider separating each class into a separate header and source file to improve readability and manageability.

3) In PlayerResources you use raw pointers to store Building objects. In modern C++, it is recommended to use smart pointers (std::unique_ptr or std::shared_ptr) to automate memory management and avoid memory leaks.
std::vector<std::unique_ptr<Building>> buildings;

**Review by staskashchenko — APPROVED** [review]
Your code fixes, incorporating changes to Building.h, separating classes into separate files, and switching to smart pointers in PlayerResources have made our code cleaner, safer, and easier to maintain. The code has improved. Thank you!

**Pull request merged**